### ISSUE TITLE
TreeSelect search doesn't update keyboard-active item

### ISSUE THREAD

**Issue by JeremyMBell** [ant-design/ant-design#50852]
### Reproduction link

[![Edit on StackBlitz](https://developer.stackblitz.com/img/open_in_stackblitz.svg)](https://stackblitz.com/edit/react-ccyi1p?file=demo.tsx)

### Steps to reproduce

1. Open the dropdown.
2. Use arrow keys to hover onto "United States of America"
3. Type "Cana"
4. Press enter
5. Observe selection.

### What is expected?

Ideally, "Canada" is selected. Alternatively, the selection isn't updated.

### What is actually happening?

"United States of America" is selected.

| Environment | Info |
| --- | --- |
| antd | 5.20.6 |
| React | 18.3.1 |
| System | Mac Sonoma 14.6.1 |
| Browser | Firefox 130.0 |

---
I wasn't sure if this is a feature request, since I could imagine this being a behavior that is controlled by props. Regardless, our use case is to select a large hierarchical list of items, such as locations.

The depth of the tree is quite large, so in order to scroll to a "Canada" using additional key presses would require us to traverse each level of the tree down to the leaf.

In my ideal scenario, the activeKey (https://github.com/react-component/tree-select/blob/master/src/OptionList.tsx#L80) would point to the first matching result.

**Comment by GeorgeHcc** [comment]
When typing in Search, ant-d does not handle the enter keyboard event when typing . so you press the enter key, the node selected by your keyboard will be used. 

This will not happen if you click on the node while typing.

**Comment by JeremyMBell** [comment]
Correct, the issue is specific to keyboard users. My opinion is that the enter key should not be able to select any option that is not visible to the user, ever.

And, I would prefer the ability to autocomplete the user's search query to improve their user experience.

**Comment by GeorgeHcc** [comment]
working in progress
